cetmix_tower_server/models/cx_tower_plan_line_action.py

```diff
@@ -60,19 +59,53 @@ def _compute_name(self):
 
             # Compose name based on values
             elif rec.condition and rec.action and rec.value_char:
-                action_string = action_selection_vals.get(rec.action)
-
-                # Add custom exit code if action presumes it
-                if rec.action == "ec":
-                    action_string = "{} {}".format(action_string, rec.custom_exit_code)
-                rec.name = " ".join(
-                    (
-                        _("If exit code"),
-                        rec.condition,
-                        rec.value_char,
-                        _("then"),
-                        action_string,
-                    )
+                rec.name = self._generate_action_name(
+                    rec.condition, rec.action, rec.value_char, rec.custom_exit_code
                 )
             else:
                 rec.name = _("Wrong action")
+
+    @api.model
+    def _generate_action_name(self, condition, action, value_char, custom_exit_code):
+        """
+        Generate the name field based on the condition, action, and value_char.
+
+        Args:
+            condition (str): The condition for the action.
+            action (str): The action to be performed.
+            value_char (str): The value associated with the action.
+            custom_exit_code (str): The custom exit code for the action.
+
+        Returns:
+            str: The generated name based on the provided parameters.
+        """
+        action_selection_vals = dict(self._fields["action"].selection)  # type: ignore
+
+        # Compose name based on values
+        if condition and action and value_char:
+            action_string = action_selection_vals.get(action)
+
+            # Add custom exit code if action presumes it
+            if action == "ec":
+                action_string = "{} {}".format(action_string, custom_exit_code)
+            return " ".join(
+                (
+                    _("If exit code"),
+                    condition,
+                    value_char,
+                    _("then"),
+                    action_string,
+                )
+            )
+
+    @api.model_create_multi
+    def create(self, vals_list):
+        for vals in vals_list:
+            name = self._generate_action_name(
+                vals.get("condition"),
+                vals.get("action"),
+                vals.get("value_char"),
+                vals.get("custom_exit_code"),
+            )
+            vals.update({"name": name if name else "Plan Line Action"})
```

Comment:
@ivs-cetmix Should be a translatable value this default value ? ("Plan Line Action")

Reply:
@

> @ivs-cetmix Should be a translatable value this default value ? ("Plan Line Action")

@GabbasovDinar nope